src/components/layer.js

```diff
@@ -0,0 +1,153 @@
+// @flow
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import MapContext from './map-context';
+import assert from '../utils/assert';
+import deepEqual from '../utils/deep-equal';
+
+import type {MapContextProps} from './map-context';
+
+const propTypes = {
+  type: PropTypes.string.isRequired,
+  id: PropTypes.string,
+  source: PropTypes.string,
+  beforeId: PropTypes.string
+};
+
+type LayerProps = {
```

Comment:
Mapox layer has other props, so will this type spec cause any flow errors if we are gonna publish flow definition?

Reply:
I don't think so: https://flow.org/try/#0C4TwDgpgBAglC8UDeAoKUCGAuKA7ArgLYBGEATigL4DcKKAbhmVAPbEBWAjFnIktlADkgygyasOAJh4JkAgAwAaKMRzDlAYxzAy+CKKA